File: Arrays/FindDuplicates/FindDuplicates.js
function method1(arr){
    let i,j,n=arr.length;
    for(i=0;i<n;i++){
        for(j=i+1;j<n;j++){
            if(arr[i]==arr[j]){
                return arr[i];
            }
        }
    }
    return 0;
}

function method2(arr){
    let slow=0, fast=0;
    while(true){
        slow = arr[slow];
        fast = arr[arr[fast]];
        if(slow==fast){
            break;
        }
    }
    slow = 0;
    while(slow!=fast){
        slow = arr[slow];
        fast = arr[fast];
    }
    return slow;
}

let arr = [1,3,4,4,2];
console.log(method1(arr));
console.log(method2(arr));

File: Arrays/FindDuplicates/FindDuplicates.go
package main

import "fmt"

func method1(arr []int) int {
	var i int
	var j int
	for i = 0; i < len(arr); i++ {
		for j = i + 1; j < len(arr); j++ {
			if arr[i] == arr[j] {
				return arr[i]
			}
		}
	}
	return 0
}

func method2(arr []int) int {
	slow := 0
	fast := 0
	for {
		slow = arr[slow]
		fast = arr[arr[fast]]
		if slow == fast {
			break
		}
	}
	slow = 0
	for slow != fast {
		slow = arr[slow]
		fast = arr[fast]
	}
	return slow
}

func main() {
	arr := []int{1, 2, 3, 4, 1}
	fmt.Println(method1(arr))
	fmt.Println(method2(arr))
}
